developer_docs/syntactic_stage.md

```diff
@@ -0,0 +1,82 @@
+# Syntactic Stage
+
+The syntactic stage is described by the file [pyccel.parser.syntactic](../pyccel/parser/syntactic.py)
+
+The syntactic stage serves 4 main purposes:
+1.  Convert Python's [AST](https://docs.python.org/3/library/ast.html) (abstract syntax tree) representation of the python file to Pyccel's AST representation (objects of the classes in the folder [pyccel.ast](../pyccel/ast))
+2.  Raise an error for any syntax used that is not yet supported by pyccel
+3.  Convert header comments from strings to Pyccel's AST representation
+4.  Collect the name of all variables in each scope (see [scope](scope.md) for more details) to ensure no name collisions can occur if pyccel generates Variable names
+
+## Navigation and AST Creation
+
+The entry point for the class `SyntaxParser` is the function `parse`.
+This function is called from the constructor to examine a FST (Full Syntax Tree) object.
+The FST is the output of Python's `ast.parse` function with comments inserted using the [`pyccel.parser.extend_tree.extend_tree`](../pyccel/parser/extend_tree.py).
+Python's ast module discards all comments, so the function [`pyccel.parser.extend_tree.extend_tree`](../pyccel/parser/extend_tree.py) is needed to read the file and reinsert these comments at the correct location.
+
+The key line of the function `parse` is the call to `self._visit(self.fst)`.
+All elements of the tree must be visited.
+The `_visit` function internally calls a function named `_visit_X`, where `X` is the type of the object.
+This function must have the form:
+```python
+def _visit_ClassName(self, stmt):
+    ...
+```
+Each of these functions should internally call the `_visit` function on each of the elements of the object to obtain Pyccel AST nodes which can be combined to create a Pyccel AST node representing the current object.
+
```

Comment:
I think we should do the same regarding nodes and their respective _visit functions. This may seem unnecessary but it will allow the redirection to the docs of all visit functions.

Reply:
> I think we should do the same 

Meaning? Write a list of all `_visit` functions? Is this not what the doc of all visit functions is for?

Reply:
Nevermind then, it stemmed from the same need to have everything in one place.